### PR TITLE
Update Deposit Helper Functions to v0.1

### LINKED ISSUE #319: Update Deposit Operations Helper Functions for Per-Block Processing
### Description
As a validator, I want helper functions for deposits processed during the per-block processing logic, so that the client is compliant with the spec and deposits are kept up to date. 

### Acceptance Criteria
Per-block:
  - Verify that `length` of `block.body.deposits` is <= `MAX_DEPOSITS`
  - For each `deposit` in `block.body.deposits`:
    - Verify that the serialized version of `deposit.deposit_data` matches the `deposit_data` in the Eth 1.0 Deposit Contract spec.
    - Verify that verify_merkle_branch of the (serialized `deposit_data` hash, deposit branch, `DEPOSIT_CONTRACT_TREE_DEPTH`, deposit index, state.latest_eth1_data deposit root) is `TRUE`.
  - Run the `process_deposit` function.

### Additional Information
https://github.com/ethereum/eth2.0-specs/blob/v0.1/specs/core/0_beacon-chain.md#deposits-1

### PR DESCRIPTION
**Pull request by schroedingerscode**
## PR Description
This PR updates helper functions associated with deposit operations of the per-block processing logic to v0.1 of the spec, and adds unit tests for those functions.

Functions updated and unit tested in `BeaconStateUtil` are:
  - `get_fork_version`
  - `get_domain`
  - `validate_proof_of_possession`  - _(the tests for this are stubs which are disabled)_
  - `process_deposit`

## Fixed Issue(s)
Resolves #319

### CODE REVIEW

File: ethereum/statetransition/src/test/java/tech/pegasys/artemis/statetransition/util/BeaconStateUtilTest.java [hunk 1]
```diff
@@ -73,4 +88,230 @@ void sqrtOfANegativeNumber() {
...
+      assertEquals(
+          BeaconStateUtil.get_domain(fork, givenEpoch, domain),
+          UnsignedLong.valueOf(
+              (BeaconStateUtil.get_fork_version(fork, givenEpoch).longValue() << 32) + domain));
+    }
+  }
+  // *************** END Fork Tests ***************
+
+  @Test
+  @Disabled
```
Comment (akhila-raju) [inline ethereum/statetransition/src/test/java/tech/pegasys/artemis/statetransition/util/BeaconStateUtilTest.java:192]: Wow I didn't know you could do this! 😱

**Review by akhila-raju — APPROVED** [review]
LGTM

File: ethereum/statetransition/src/test/java/tech/pegasys/artemis/statetransition/util/BeaconStateUtilTest.java [hunk 2]
```diff
@@ -73,4 +88,230 @@ void sqrtOfANegativeNumber() {
           BeaconStateUtil.integer_squareroot(UnsignedLong.valueOf(-1L));
         });
   }
+
+  // TODO It may make sense to move these tests to a Fork specific test class in the future.
+  // *************** START Fork Tests ***************
+  void getForkVersionReturnsPreviousVersionWhenGivenEpochIsLessThanForkEpoch() {
```
Comment (jrhea) [inline ethereum/statetransition/src/test/java/tech/pegasys/artemis/statetransition/util/BeaconStateUtilTest.java:95]: Do you need an `@Test` ?

**Review by jrhea — APPROVED** [review]
Just one comment.  I trust you will make the change if it needs to be made.

File: ethereum/statetransition/src/test/java/tech/pegasys/artemis/statetransition/util/BeaconStateUtilTest.java (hunk 2 above)
Comment (schroedingerscode) [reply ethereum/statetransition/src/test/java/tech/pegasys/artemis/statetransition/util/BeaconStateUtilTest.java:95]: Good catch. Fixed.

**Pull request merged**